cdap-security/src/main/java/co/cask/cdap/security/tools/GeneratedCertKeyStoreCreator.java

```diff
@@ -0,0 +1,169 @@
+                                   "a password in cdap-security.xml using " +
+                                   Constants.Security.AppFabric.SSL_KEYSTORE_PASSWORD);
+    }
+    try {
+      String keyPairAlgorithm = sConf.get(Constants.Security.AppFabric.KEY_PAIR_ALGORITHM,
+                                          KEY_PAIR_ALGORITHM);
+      KeyPairGenerator keyGen = KeyPairGenerator.getInstance(keyPairAlgorithm);
+      String randomAlgorithm = sConf.get(Constants.Security.AppFabric.SECURE_RANDOM_ALGORITHM,
+                                         SECURE_RANDOM_ALGORITHM);
+      String randomProvider = sConf.get(Constants.Security.AppFabric.SECURE_RANDOM_PROVIDER,
+                                        SECURE_RANDOM_PROVIDER);
+      SecureRandom random = SecureRandom.getInstance(randomAlgorithm, randomProvider);
+      keyGen.initialize(KEY_SIZE, random);
+      // generate a key pair
+      KeyPair pair = keyGen.generateKeyPair();
+      int validity = sConf.getInt(Constants.Security.AppFabric.CERT_VALIDITY, VALIDITY);
+      String distinguishedName = sConf.get(Constants.Security.AppFabric.CERT_DISTINGUISHED_NAME, DISTINGUISHED_NAME);
+      String signatureAlgo = sConf.get(Constants.Security.AppFabric.SIGNATURE_ALGORITHM, SIGNATURE_ALGORITHM);
+
+      X509Certificate cert = getCertificate(distinguishedName, pair, validity, signatureAlgo);
+      InputStream inputStream = new ByteArrayInputStream(cert.getEncoded());
+
+      keyStore = KeyStore.getInstance(sConf.get(Constants.Security.AppFabric.SSL_KEYSTORE_TYPE, SSL_KEYSTORE_TYPE));
+      keyStore.load(null, password.toCharArray());
+      keyStore.setKeyEntry(CERT_ALIAS, pair.getPrivate(), password.toCharArray(),
+                           new java.security.cert.Certificate[]{cert});
+    } catch (Throwable e) {
+      throw new RuntimeException("SSL is enabled but a key store file could not be created. A keystore is required " +
+                                   "for SSL to be used.", e);
+    }
+    return keyStore;
+  }
+
+  /**
+   * Generate an X.509 certificate
+   * @param dn Distinguished name for the owner of the certificate, it will also be the signer of the certificate.
+   * @param pair Key pair used for signing the certificate.
+   * @param days Validity of the certificate.
+   * @param algorithm Name of the signature algorithm used.
+   * @return A X.509 certificate
+   */
+  private static X509Certificate getCertificate(String dn, KeyPair pair, int days, String algorithm) {
+    // Calculate the validity interval of the certificate
+    Date from = new Date();
+    Date to = DateUtils.addDays(from, days);
+    CertificateValidity interval = new CertificateValidity(from, to);
+    // Generate a random number to use as the serial number for the certificate
+    BigInteger sn = new BigInteger(64, new SecureRandom());
+    X509CertInfo info = null;
+    X509CertImpl cert = null;
```

Comment:
same here, and then you can return the cert from the ``try`` block, since the ``catch`` throws a ``RuntimeException``?